logging/WFCORE-4807-log-routing.adoc

```diff
@@ -0,0 +1,112 @@
+= Log Routing
+:author:            James R. Perkins
+:email:             jperkins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently any modules that log generally log onto the system log context. In some cases, for example a deployment using
+a logging profile, the loggers may end up getting created on the wrong `LogContext`. As an example if `a.war` and `b.war`
+both depend on module `org.jboss.custom.module` and at least one deployment uses a logging profile whichever deployment
+initializes the custom module first will have the loggers created on that `LogContext`. If a logging profile is being
+used it would be that profiles log context. This means if `a.war` uses a logging profile and the module is accessed first
+by `a.war`, then static loggers will be created on the logging profiles `LogContext`. Once `b.war` invokes something to
+be logged in the custom module, then `b.war` 's log messages end up in the logging profiles `LogContext` instead of the
+systems `LogContext`.
+
+https://issues.jboss.org/browse/LOGMR-264[LOGMR-264] will introduce a way to route log messages based on the log context.
+This would allow messages from modules to be routed to deployment specific log contexts. It would also introduce a
+change in behavior which is the reason for an RFE. A new attribute, `allow-log-routing` will be introduced to disable
+the new routing and allow the old behavior.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4897[WFCORE-4807]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1420[EAP7-1420]
+* https://issues.jboss.org/browse/LOGMR-264[LOGMR-264]
+* https://issues.redhat.com/browse/WFCORE-3376[WFCORE-3376]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mkopecky@redhat.com[Marek Kopeck&#253;]
+
+=== Testing By
+
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+* JBoss Log Manager
+
+=== Other Interested Projects
+
+* Any consumer of WildFly Core that uses the logging subsystem.
+
+== Requirements
+
+=== Hard Requirements
+
+* The attribute, `allow-log-routing`, is a simple boolean attribute which determines if the new or old behavior will be used.
```

Comment:
Please explicitly specify location of `allow-log-routing`attribute in management model and default value.